### PR TITLE
[#34][UI]As a user, I can register an account 

### PR DESCRIPTION
**Pull request by mosharaf13**
Closes https://github.com/mosharaf13/google-scrapper-ruby/issues/34

## What happened 👀

- Added styles to the user registration form

## Insight 📝

- Added bootstrap form classes to make the user registration form more user friendly.

## Proof Of Work 📹
<details>
<summary>
New form
</summary>

Screen Shot 2023-05-25 at 5 04 06 PM
</details>

<details>
<summary>
New form with error alert
</summary>

Screen Shot 2023-05-25 at 5 05 21 PM
</details>

### CODE REVIEW

File: app/views/devise/passwords/edit.html.erb [hunk 1]
```diff
@@ -0,0 +1,25 @@
+<h2>Change your password</h2>
+
+<%= form_for(resource, as: resource_name, url: password_path(resource_name), html: { method: :put }) do |f| %>
+  <%= render "devise/shared/error_messages", resource: resource %>
```
Comment (sanG-github) [inline app/views/devise/passwords/edit.html.erb:4]: We usually use single quote ' for Ruby code (start with <% or <%=)
And we use double quotes " for HTML code

```suggestion
  <%= render 'devise/shared/error_messages', resource: resource %>
```

File: app/views/devise/passwords/edit.html.erb [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+  <%= f.hidden_field :reset_password_token %>
+
+  <div class="field">
+    <%= f.label :password, "New password" %><br />
```
Comment (sanG-github) [inline app/views/devise/passwords/edit.html.erb:8]: The same suggestion here, please update other similar places  🙏 
```suggestion
    <%= f.label :password, 'New password' %><br />
```

File: db/schema.rb [hunk 3]
```diff
@@ -10,12 +10,23 @@
...
 # It's strongly recommended that you check this file into your version control system.
 
-ActiveRecord::Schema.define(version: 2023_05_22_073804) do
+ActiveRecord::Schema.define(version: 2023_05_23_085326) do
 
   # These are extensions that must be enabled in order to support this database
   enable_extension "citext"
   enable_extension "plpgsql"
 
+  create_table "search_stats", force: :cascade do |t|
```
Comment (sanG-github) [inline db/schema.rb:19]: Please rebase with the develop branch 🙏

**Review by sanG-github — COMMENTED** [review]
Please make the `log in` button to have the same style as the `sign up` button.

image

File: app/views/devise/passwords/edit.html.erb (hunk 2 above)
Comment (mosharaf13) [reply app/views/devise/passwords/edit.html.erb:8]: Okay 👍🏼

File: db/schema.rb (hunk 3 above)
Comment (mosharaf13) [reply db/schema.rb:19]: Done 👍🏼

File: app/views/devise/passwords/edit.html.erb (hunk 1 above)
Comment (longnd) [reply app/views/devise/passwords/edit.html.erb]: @mosharaf13 to give more context, it is not a habit that we usually do, but it is a convention
* https://nimblehq.co/compass/development/code-conventions/html/#convention-2
* https://nimblehq.co/compass/development/code-conventions/ruby/#convention-29

File: app/views/devise/mailer/unlock_instructions.html.erb [hunk 4]
```diff
@@ -0,0 +1,7 @@
+<p>Hello <%= @resource.email %>!</p>
```
[the remark this answers is not in the document]
Comment (mosharaf13) [reply app/views/devise/mailer/unlock_instructions.html.erb]: Thanks. Cleaned all unnecessary devise related files.

File: db/schema.rb (hunk 3 above)
[the remark this answers is not in the document]
Comment (mosharaf13) [reply db/schema.rb:19]: Right. Removing.

File: app/views/devise/registrations/new.html.erb [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+        <div class="field form-group">
+          <%= f.label :email %><br />
+          <%= f.email_field :email, autofocus: true, autocomplete: 'email', class: 'form-control' %>
+        </div>
+
+        <div class="field form-group mt-3">
+          <%= f.label :password %>
+          <% if @minimum_password_length %>
+            <em>(<%= @minimum_password_length %> characters minimum)</em>
+          <% end %><br />
```
Comment (sanG-github) [inline app/views/devise/registrations/new.html.erb:18]: As we've configured it in `config/initializers/devise.rb`, then should we remove this check? 😉 
```suggestion
          <em>(<%= @minimum_password_length %> characters minimum)</em>
```

**Review by sanG-github — APPROVED** [review]
We're getting there!!! 🚀 

But from my perspective, it's not clear enough for the instruction of the password. As the screen shown me that the password should be at least 8 characters, but it's not all for the validation. So, I suggest removing the instruction, and then add the locale for the `invalid` to describe which is the password pattern means.
- Contains at least one uppercase letter.
- Contains at least one lowercase letter.
- Contains at least one digit.
- Has a minimum length of 8 characters.

```ruby
PASSWORD_PATTERN = /(?=.*[A-Z])(?=.*[a-z])(?=.*\d).{8,}/
```
image


But, it's a minor thing about the UX, feel free to challenge yourself. 😉

File: app/views/devise/registrations/new.html.erb (hunk 5 above)
Comment (mosharaf13) [reply app/views/devise/registrations/new.html.erb]: Fixed in d8b5acf34a6b61e2e721ed26dfc044bfcdadabb7

File: app/views/devise/passwords/edit.html.erb (hunk 1 above)
Comment (mosharaf13) [reply app/views/devise/passwords/edit.html.erb]: Thanks 👍🏼 . Fixed in f32a21959398bd0a548afa6fd3ef0cfbd927e674

File: app/views/devise/mailer/unlock_instructions.html.erb (hunk 4 above)
[the remark this answers is not in the document]
Comment (mosharaf13) [reply app/views/devise/mailer/unlock_instructions.html.erb]: Fixed in 417770e894aa02827d194277e228c58d084b3ea0

File: db/schema.rb (hunk 3 above)
[the remark this answers is not in the document]
Comment (mosharaf13) [reply db/schema.rb]: Fixed in fb67d2110936b1316002c7d7e5ff6c80b5b4a8aa

**Review by longnd — APPROVED** [review]

**Pull request merged**